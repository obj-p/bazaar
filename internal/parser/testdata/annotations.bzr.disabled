package annotations

func MultipleAnnotations() {
    // Multiple annotations work on function calls
    @State @Modifier(Padding(all = 42)) @Custom
    SomeFunction()

    // Single annotations work on both calls and variables
    @State
    AnotherFunction()

    @State
    var singleVar = 1

    // Multiple annotations work on function calls with parameters
    @Logger @Async @Retry
    ProcessData("input", param2 = 42)
}
