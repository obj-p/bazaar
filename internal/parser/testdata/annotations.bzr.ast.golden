&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "annotations",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Function: &parser.FunctionDecl{
        Name: "MultipleAnnotations",
        Block: []*parser.Stmt{
          {
            Callable: &parser.CallableExpr{
              Annotations: []*parser.AnnotationExpr{
                {
                  Name: &"State",
                },
                {
                  Name: &"Modifier",
                  Call: &parser.CallExpr{
                    Arguments: []*parser.ArgumentExpr{
                      {
                        Expr: &parser.Expr{
                          Unary: &parser.UnaryExpr{
                            Primary: &parser.PrimaryExpr{
                              Callable: &parser.CallableExpr{
                                Name: &"Padding",
                                Arguments: []*parser.ArgumentExpr{
                                  {
                                    Name: &"all",
                                    Expr: &parser.Expr{
                                      Unary: &parser.UnaryExpr{
                                        Primary: &parser.PrimaryExpr{
                                          Literal: &parser.Literal{
                                            Number: &"42",
                                          },
                                        },
                                      },
                                    },
                                  },
                                },
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
                {
                  Name: &"Custom",
                },
              },
              Name: &"SomeFunction",
            },
          },
          {
            Callable: &parser.CallableExpr{
              Annotations: []*parser.AnnotationExpr{
                {
                  Name: &"State",
                },
              },
              Name: &"AnotherFunction",
            },
          },
          {
            Var: &parser.VarDeclStmt{
              Annotations: []*parser.AnnotationExpr{
                {
                  Name: &"State",
                },
              },
              Dest: &parser.VariableExpr{
                Name: &"singleVar",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"1",
                    },
                  },
                },
              },
            },
          },
          {
            Callable: &parser.CallableExpr{
              Annotations: []*parser.AnnotationExpr{
                {
                  Name: &"Logger",
                },
                {
                  Name: &"Async",
                },
                {
                  Name: &"Retry",
                },
              },
              Name: &"ProcessData",
              Arguments: []*parser.ArgumentExpr{
                {
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          String: &parser.String{
                            Fragments: []*parser.StringFragment{
                              {
                                Text: &"input",
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
                {
                  Name: &"param2",
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          Number: &"42",
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
